From 732fc1f52151dfbe5a0b9ebe5c04f8f7385cdb7b Mon Sep 17 00:00:00 2001
From: Ola Jeppsson <ola@adapteva.com>
Date: Thu, 7 May 2015 22:53:37 +0000
Subject: [PATCH] boards/adapteva/parallella: Embed configuration in
 fw_setenv/fw_printenv

This patch embeds Parallella mtd flash configuration in the
fw_*env binaries so there we don't need to ship a separate
configuartion file.

make adapteva_parallella_defconfig && make env

Symlink fw_setenv to fw_printenv.

Signed-off-by: Ola Jeppsson <ola@adapteva.com>
---
 tools/env/fw_env.h | 22 +++++++++++-----------
 1 file changed, 11 insertions(+), 11 deletions(-)

diff --git a/tools/env/fw_env.h b/tools/env/fw_env.h
index d6faf34..498c3ac 100644
--- a/tools/env/fw_env.h
+++ b/tools/env/fw_env.h
@@ -21,20 +21,20 @@
  * See included "fw_env.config" sample file
  * for notes on configuration.
  */
-#define CONFIG_FILE     "/etc/fw_env.config"
+/* #define CONFIG_FILE     "/etc/fw_env.config" */
 
 #ifndef CONFIG_FILE
-#define HAVE_REDUND /* For systems with 2 env sectors */
-#define DEVICE1_NAME      "/dev/mtd1"
-#define DEVICE2_NAME      "/dev/mtd2"
+#undef HAVE_REDUND /* For systems with 2 env sectors */
+#define DEVICE1_NAME      "/dev/mtdblock1"
+/* #define DEVICE2_NAME      "/dev/mtd2" */
 #define DEVICE1_OFFSET    0x0000
-#define ENV1_SIZE         0x4000
-#define DEVICE1_ESIZE     0x4000
-#define DEVICE1_ENVSECTORS     2
-#define DEVICE2_OFFSET    0x0000
-#define ENV2_SIZE         0x4000
-#define DEVICE2_ESIZE     0x4000
-#define DEVICE2_ENVSECTORS     2
+#define ENV1_SIZE         0x20000
+#define DEVICE1_ESIZE     0x20000
+#define DEVICE1_ENVSECTORS     1
+/* #define DEVICE2_OFFSET    0x0000 */
+/* #define ENV2_SIZE         0x4000 */
+/* #define DEVICE2_ESIZE     0x4000 */
+/* #define DEVICE2_ENVSECTORS     2 */
 #endif
 
 #ifndef CONFIG_BAUDRATE
-- 
1.9.1

